File: solutions/python/spiral-matrix/1/spiral_matrix.py
def spiral_matrix(size):
    n = size  # Assign size to n for convenience
    matrix = [[0] * n for _ in range(n)]

    left, right = 0, n - 1
    top, bottom = 0, n - 1
    num = 1

    while left <= right and top <= bottom:
        # Traverse from left to right
        for i in range(left, right + 1):
            matrix[top][i] = num
            num += 1
        top += 1

        # Traverse from top to bottom
        for i in range(top, bottom + 1):
            matrix[i][right] = num
            num += 1
        right -= 1

        # Traverse from right to left
        if top <= bottom:
            for i in range(right, left - 1, -1):
                matrix[bottom][i] = num
                num += 1
            bottom -= 1

        # Traverse from bottom to top
        if left <= right:
            for i in range(bottom, top - 1, -1):
                matrix[i][left] = num
                num += 1
            left += 1

    return matrix

File: solutions/javascript/spiral-matrix/1/spiral-matrix.js
//
// This is only a SKELETON file for the 'Spiral Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const spiralMatrix = (size) => {
let n=size;
let matrix = [];
for (let i = 0; i < n; i++) {
  matrix.push(Array(n).fill(0));
}
let left = 0, right = n - 1;
let top = 0, bottom = n - 1;
let num = 1;

 while (left <= right && top <= bottom) {
   for (let i = left; i <= right; i++) {
  matrix[top][i] = num;
  num++;
}
top++;
   for (let i = top; i <= bottom; i++) {
  matrix[i][right] = num;
  num++;
}
right--;
  if (top <= bottom) {
    for (let i = right; i >= left; i--) {
      matrix[bottom][i] = num;
      num++;
    }
    bottom--;
  }

  
  if (left <= right) {
    for (let i = bottom; i >= top; i--) {
      matrix[i][left] = num;
      num++;
    }
    left++;
  }
}

return matrix;
};
